src/dkg/round1.rs

```diff
@@ -123,14 +134,119 @@ pub fn import_secret_package(
     SerializableSecretPackage::deserialize_from(&serialized[..]).map(|pkg| pkg.into())
 }
 
+#[must_use]
+fn input_checksum<I>(min_signers: u16, signing_participants: &[I]) -> Checksum
+where
+    I: Borrow<Identity>,
+{
+    let mut hasher = ChecksumHasher::new();
+
+    hasher.write(&min_signers.to_le_bytes());
+
+    let mut signing_participants = signing_participants
+        .iter()
+        .map(Borrow::borrow)
+        .collect::<Vec<_>>();
+    signing_participants.sort_unstable();
+    signing_participants.dedup();
+
+    for id in signing_participants {
+        hasher.write(&id.serialize());
+    }
+
+    hasher.finish()
+}
+
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub struct PublicPackage {
+    identity: Identity,
+    frost_package: Package,
+    group_secret_key_shard: [u8; 32],
```

Comment:
I would change this to be `GroupSecretKeyShard`

Reply:
Ack. The GroupSecretKeyShard will need to be encrypted anyway, so this is only temporary.